dev/NavigationView/NavigationView.cpp

```diff
@@ -3632,3 +3820,159 @@ void NavigationView::UpdatePaneShadow()
         shadowReceiver.Margin(shadowReceiverMargin);
     }
 }
+
+template<typename T> T NavigationView::GetContainerForData(const winrt::IInspectable& data)
+{
+    if (!data)
+    {
+        return nullptr;
+    }
+
+    if (auto nvi = data.try_as<T>())
+    {
+        return nvi;
+    }
+
+    if (IsTopNavigationView())
+    {
+        auto ir = m_topNavRepeater.get();
+        auto itemIndex = m_topDataProvider.IndexOf(data, NavigationViewSplitVectorID::PrimaryList);
+        if (itemIndex >= 0)
+        {
+            if (auto container = ir.TryGetElement(itemIndex))
+            {
+                return container.try_as<T>();
+            }
+        }
+    }
+    else
+    {
+        auto ir = m_leftNavRepeater.get();
+        int indexOfData = LeftNavGetIndexFromItem(data);
+        if (indexOfData > -1 && ir)
+        {
+            if (auto container = ir.TryGetElement(indexOfData))
+            {
+                return container.try_as<T>();
+            }
+        }
+    }
+
+    if (auto settingsItem = m_settingsItem.get())
+    {
+        if (settingsItem == data || settingsItem.Content() == data)
+        {
+            return settingsItem.try_as<T>();
+        }
+    }
+
+    return nullptr;
+}
+
+int NavigationView::LeftNavGetIndexFromItem(const winrt::IInspectable& data)
+{
+    if (m_leftNavRepeater)
+    {
+        winrt::ItemsSourceView dataSourceView = m_leftNavRepeater.get().ItemsSourceView();
+        if (dataSourceView)
+        {
+            auto inspectingDataSource = static_cast<InspectingDataSource*>(winrt::get_self<ItemsSourceView>(dataSourceView));
+            return inspectingDataSource->IndexOf(data);
+        }
+    }
+    return -1;
+}
+
+winrt::IInspectable NavigationView::LeftNavGetItemFromIndex(int index)
+{
+    if (m_leftNavRepeater)
+    {
+        winrt::ItemsSourceView dataSourceView = m_leftNavRepeater.get().ItemsSourceView();
+        if (dataSourceView)
+        {
+            return dataSourceView.GetAt(index);
+        }
+    }
+    return nullptr;
+}
+
+winrt::NavigationViewItemBase NavigationView::GetContainerForIndexPath(const winrt::IndexPath& indexPath)
```

Comment:
>indexPath [](start = 95, length = 9)

why is this an index path if we only ever read the first value?

Reply:
Primarily this is in anticipation of hierarchical nav which will require the use of the indexpath returned by SelectionModel